src/main/java/net/spy/memcached/protocol/BaseOperationImpl.java

```diff
@@ -85,6 +86,7 @@
   public BaseOperationImpl() {
     super();
     creationTime = System.nanoTime();
+    stateObservers = new ArrayList<OperationStateChangeObserver>();
```

Comment:
ArrayDeque is faster in this case?

Reply:
why?

Reply:
Not sure it's still relevant:
https://publicobject.com/2010/07/07/caliper_confirms_reality_linked_list_vs_array_list/